$MirOS: ports/www/firesomething/patches/patch-configure_in,v 1.3 2008/03/31 21:33:56 bsiegert Exp $
$OpenBSD: patch-configure_in,v 1.5 2005/04/09 08:11:19 wilfried Exp $
--- configure.in.orig	Wed Aug 23 18:30:40 2006
+++ configure.in	Mon Mar 31 21:47:50 2008
@@ -42,7 +42,7 @@ dnl ====================================
 
 AC_PREREQ(2.13)
 AC_INIT(config/config.mk)
-AC_CONFIG_AUX_DIR(${srcdir}/build/autoconf)
+AC_CONFIG_AUX_DIR(build/autoconf)
 AC_CANONICAL_SYSTEM
 TARGET_CPU="${target_cpu}"
 TARGET_VENDOR="${target_vendor}"
@@ -694,7 +694,7 @@ tools are selected during the Xcode/Deve
                  int main() { return 0; }],
    result=yes,
    result=no)
-  AC_LANG_RESTORE
+  AC_LANG_C
   AC_MSG_RESULT($result)
 
   if test "$result" = "no" ; then
@@ -1291,9 +1291,9 @@ dnl Get mozilla version from central mil
 MOZILLA_VERSION=`$PERL $srcdir/config/milestone.pl -topsrcdir $srcdir`
 
 dnl Get version of various core apps from the version files.
-FIREFOX_VERSION=`cat $topsrcdir/browser/config/version.txt`
-THUNDERBIRD_VERSION=`cat $topsrcdir/mail/config/version.txt`
-SEAMONKEY_VERSION=`cat $topsrcdir/xpfe/bootstrap/version.txt`
+FIREFOX_VERSION=`cat ./browser/config/version.txt`
+THUNDERBIRD_VERSION=`cat ./mail/config/version.txt`
+SEAMONKEY_VERSION=`cat ./xpfe/bootstrap/version.txt`
 
 AC_DEFINE_UNQUOTED(MOZILLA_VERSION,"$MOZILLA_VERSION")
 AC_DEFINE_UNQUOTED(MOZILLA_VERSION_U,$MOZILLA_VERSION)
@@ -1862,8 +1862,8 @@ case "$target" in
 	esac
 	;;
 
-*-openbsd*)
-    DLL_SUFFIX=".so.1.0"
+*-openbsd*|*-mirbsd*|*-midnightbsd*)
+    DLL_SUFFIX=".so.${SO_VERSION}"
     DSO_CFLAGS=''
     DSO_PIC_CFLAGS='-fPIC'
     DSO_LDOPTS='-shared -fPIC'
@@ -2726,7 +2726,7 @@ then
 			fi
 			;;
 
-	    *-*-openbsd*|*-*-bsdi*)
+	    *-*-openbsd*|*-*-bsdi*|*-*-mirbsd*|*-*-midnightbsd*)
 			AC_DEFINE(_REENTRANT)
 			AC_DEFINE(_THREAD_SAFE)
 			dnl -pthread links in -lc_r, so don't specify it explicitly.
@@ -3647,7 +3647,7 @@ _SAVE_LIBS=$LIBS
 CFLAGS="$ZLIB_CFLAGS $CFLAGS"
 LDFLAGS="$ZLIB_LIBS -lz $LDFLAGS"
 if test -n "${PNG_DIR}" -a "${PNG_DIR}" != "yes"; then
-    CFLAGS="-I${PNG_DIR}/include $CFLAGS"
+    CFLAGS="-I${PNG_DIR}/include/libpng $CFLAGS"
     LDFLAGS="-L${PNG_DIR}/lib $LDFLAGS"
 fi
 if test -z "$PNG_DIR" -o "$PNG_DIR" = no; then
@@ -3677,7 +3677,7 @@ LDFLAGS=$_SAVE_LDFLAGS
 LIBS=$_SAVE_LIBS
 
 if test "${PNG_DIR}" -a -d "${PNG_DIR}" -a "$SYSTEM_PNG" = 1; then
-    PNG_CFLAGS="-I${PNG_DIR}/include"
+    PNG_CFLAGS="-I${PNG_DIR}/include/libpng"
     PNG_LIBS="-L${PNG_DIR}/lib ${PNG_LIBS}"
 fi
 
